app/lib/identity_tijuana/member_sync.rb

```diff
@@ -78,6 +147,88 @@ def self.import_user(user_id, sync_id)
       calling_subscription: ['associated_audits', 'Member_subscription', %w[unsubscribed_at]],
     }.freeze
 
+    # Find the primary TJ user match for a given ID member.
+    #
+    # Looks at the user for each given id, orders them by created at
+    # date, then selects:
+    #
+    #  1. The user with a normalised (but not cleaned), matching email, if any
+    #  2. The first subscribed user, if any
+    #  3. The first remaining user
+    def self.find_primary_user_for_member(member)
+      external_ids = member.member_external_ids.where(system: 'tijuana')
+      if external_ids.nil? || external_ids.count == 0
+        raise "Id member has no linked TJ users"
+      end
+
+      # Some scenarios to think about before changing this:
```

Comment:
Yeah, that's why the comment is "some scenarios to think about" rather than "here is a complete list of scenarios".

We need to get the former right for things we care about, but also the latter is either in the worst case infinitely long (but I don't think so) or in the best case too big and too complex to fit into a comment in the source code.

So instead of trying to do the latter, these are some provocations to get people to think a bit about the problem space before changing the code.

> It's unclear, which user was created first, and whether the typo'ed user got cleaned/recreated by Id as part of sync process.

So what's your understanding of the n different permutations of that?

Reply:
My question wasn't about permutations, yes there are a number of them, the root cause being that TJ is allowing these records to be created shifting the responsibility to Id to bring them in sync.

I guess, I carried into this question the thought that by determining `primary` link in this call, we are eliminating other users from being syncd, thus the question I had was
> Isn't the change of postcode valid change to be synced, regardless of which linked user was modified?

Reply:
> Isn't the change of postcode valid change to be synced, regardless of which linked user was modified

The answer is of course yes, but doing so correctly would require a three-way merge with both the Id member and the primary TJ user, and see my comments elsewhere about wanting to get something better in that what we currently have sooner rather than later.